src/Maybe.js

```diff
@@ -20,7 +20,7 @@ util.extend(Nothing, Maybe);
 Nothing.prototype.isNothing = true;
 Nothing.prototype.isJust = false;
 
-var _nothing = new Nothing();
+var _nothing = Object.freeze(new Nothing());
```

Comment:
> Is Object.freeze available in all of them?

I considered that. This looks good enough to me: http://caniuse.com/#search=freeze

i like the idea that there is one Nothing. Many Nothings feels pretty strange. (On reflection, *one* Nothing is a little strange.) But that is another possible solution.